scheduler/src/cook/rest/api.clj

```diff
@@ -182,7 +182,8 @@
   ; preemption - checkpoint is created on preemption before the VM is stopped
   {:mode (s/enum "auto" "periodic" "preemption")
    (s/optional-key :options) CheckpointOptions
-   (s/optional-key :periodic-options) PeriodicCheckpointOptions})
+   (s/optional-key :periodic-options) PeriodicCheckpointOptions
+   (s/optional-key :memory-overhead) s/Num})
```

Comment:
Why are we exposing this in the API?

Reply:
I was thinking to allow for maximum flexibility in case the checkpointing behavior is highly variable.

If this ends up not being used and/or is removed later, would you rather not add it in at all?

Reply:
Yeah, I don't think this is appropriate for users to set.

Reply:
Even if the memory used for checkpointing is a function of the job, shouldn't we be able to define that function rather than asking the user for the answer?

Reply:
removed